services/job_subscriber.go

```diff
@@ -103,20 +103,19 @@ func (js *jobSubscriber) Disconnect() {
 
 // OnNewHead resumes all pending job runs based on the new head activity.
 func (js *jobSubscriber) OnNewHead(head *models.Head) {
-	pendingRuns, err := js.store.UnscopedJobRunsWithStatus(models.RunStatusPendingConfirmations)
-	if err != nil {
-		logger.Error("error fetching pending job runs:", err.Error())
-	}
-
 	height := head.ToInt()
-	logger.Debugw("Received new head",
-		"current_height", height,
-		"pending_run_count", len(pendingRuns),
-	)
-	for _, jr := range pendingRuns {
-		err := ResumeConfirmingTask(&jr, js.store.Unscoped(), height)
+	logger.Debugw("Received new head", "current_height", height)
+
+	err := js.store.UnscopedJobRunsWithStatus(func(run *models.JobRun) error {
+		err := ResumeConfirmingTask(run, js.store.Unscoped(), height)
 		if err != nil {
-			logger.Error("JobSubscriber.OnNewHead: ", err.Error())
+			logger.Errorf("JobSubscriber.OnNewHead: %v", err)
 		}
+
+		return nil
```

Comment:
Optional: small concern here around developer usability. Returning an error in the `callback cb` will abort the entire batch, which seems easy to stumble upon if you tried:

```golang
	err := js.store.UnscopedJobRunsWithStatus(func(run *models.JobRun) error {
			return ResumeConfirmingTask(run, js.store.Unscoped(), height) // uh oh

```

Only solution I can think of is the `bool` approach in jobs but that isn't much better.

Reply:
Agree and it wasn't being used so removed.